ee/clickhouse/util.py

```diff
@@ -78,3 +82,25 @@ def tearDown(self):
         sync_execute(EVENTS_TABLE_SQL)
         sync_execute(DROP_PERSON_TABLE_SQL)
         sync_execute(PERSONS_TABLE_SQL)
+
+
+def snapshot_clickhouse_queries(fn):
+    """
+    Captures and snapshots select queries from test using `syrupy` library.
+
+    Requires queries to be stable to avoid flakiness.
+
+    Snapshots are automatically saved in a __snapshot__/*.ambr file.
+    Update snapshots via --update-snapshots.
+    """
+
+    @wraps(fn)
+    def wrapped(self, *args, **kwargs):
+        with self.capture_select_queries() as queries:
+            fn(self, *args, **kwargs)
+
+        for query in queries:
```

Comment:
(Aside: Can I get the docs to saving multiple snapshots for a single test? Didn't find it on quick glance at syrupy)

A test can have multiple queries, does this snapshotting store multiple values if necessary?

Reply:
It does - you can actually see it in some places in one of my PRs. It stats appending `.1`, `.2` etc to the snapshot name.

Reply:
Example: https://github.com/PostHog/posthog/pull/6904/files#diff-4d3a7764d124ac1edf998e86a264ea81ac45b7761f819f72e13af10b2f8591fcR80